### PR TITLE
Feature Telegram Notification

### PR DESCRIPTION
**Pull request by C-SELLERS**
#### Description

Add Telegram Notification piping through notification manager and message handler

#### Related Issue

N/A

#### Motivation and Context

Expanding notification options 

#### Requires Documentation Change

Will require documentation covering the usage

#### How Has This Been Tested?

This PR is a WIP

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Common/Notifications/NotificationManager.cs [hunk 1]
```diff
@@ -122,6 +122,24 @@ public bool Web(string address, object data = null, Dictionary<string, string> h
             return true;
         }
 
+        /// <summary>
+        /// Send a telegram message to the user specified
+        /// </summary>
+        /// <param name="user">telegram user to send to</param>
+        /// <param name="message">Message to send</param>
+        public bool Telegram(string user, string message)
```
Comment (Martin-Molinero) [inline Common/Notifications/NotificationManager.cs:130]: Believe it should allow specifying a token too?

File: Tests/Common/Notifications/NotificationJsonConverterTests.cs [hunk 2]
```diff
@@ -77,6 +77,20 @@ public void WebRoundTrip(bool nullFields)
...
+        public void TelegramRoundTrip(bool nullFields)
+        {
+            var expected = new NotificationTelegram("pepe", nullFields ? null : "ImAMessage");
+
+            var serialized = JsonConvert.SerializeObject(expected);
+
+            var result = (NotificationTelegram)JsonConvert.DeserializeObject<Notification>(serialized);
+
+            Assert.AreEqual(expected.User, result.User);
+            Assert.AreEqual(expected.Message, result.Message);
```
Comment (Martin-Molinero) [inline Tests/Common/Notifications/NotificationJsonConverterTests.cs:91]: Should add a case with and without `token` set 👍

**Review by Martin-Molinero — COMMENTED** [review]
Cool! Minor requests 👍

File: Common/Notifications/NotificationJsonConverter.cs [hunk 3]
```diff
@@ -74,6 +74,12 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
 
                 return new NotificationWeb(token.ToString(), data?.ToString(), headers?.ToObject<Dictionary<string, string>>());
             }
+            else if (jObject.TryGetValue("User", StringComparison.InvariantCultureIgnoreCase, out token))
```
[the remark this answers is not in the document]
Comment (C-SELLERS) [reply Common/Notifications/NotificationJsonConverter.cs:77]: var token is just a JToken 👌🏽 to extract the value, used to determine which type of notification is being converted. Could do inline out declaration and call it user but figured it fit in with the rest this way 😁

[the remark this answers is not in the document]
Comment (Martin-Molinero) [reply Common/Notifications/NotificationJsonConverter.cs:77]: oh lol! I see it now 👍

[the remark this answers is not in the document]
Comment (Martin-Molinero) [reply Common/Notifications/NotificationJsonConverter.cs:77]: Still missing to deserialize the actual telegram `token` if available?

**Review by Martin-Molinero — APPROVED** [review]

File: Common/Notifications/NotificationJsonConverter.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (C-SELLERS) [reply Common/Notifications/NotificationJsonConverter.cs]: Its there 😁 I think this just is an outdated file version 

https://github.com/QuantConnect/Lean/pull/5535/files#diff-c694e1def30fbfbea61f96b024990a21081c6b41a4a8fc5eb32bfd8781575a6cR80

**Pull request merged**